### PR TITLE
Added redirect to public_url

### PR DESCRIPTION
**Pull request by EGiataganas**
### Description of change

Added redirect to council's public_url

### BEFORE DEPLOYING
- we need to add the new environment variable on AWS
- we need to change `APPLICANTS_APP_HOST` environment variable on AWS

### Story Link

https://trello.com/c/K3mnhPvO

### Tasks

- Change the environment variable `APPLICANTS_APP_HOST` to map to the new one
- Add the environment variable `PUBLIC_URL`

### CODE REVIEW

File: app/models/planning_application.rb [hunk 1]
```diff
@@ -261,7 +261,15 @@ def flagged_proposal_details(flag)
   def secure_change_url
     protocol = Rails.env.production? ? "https" : "http"
 
-    "#{protocol}://#{local_authority.subdomain}.#{ENV['APPLICANTS_APP_HOST']}/validation_requests?planning_application_id=#{id}&change_access_id=#{change_access_id}"
+    ENV.fetch("PUBLIC_URL", "false") == "true" ? "#{protocol}://#{public_url}" : "#{protocol}://#{internal_url}"
+  end
+
+  def public_url
```
Comment (benbaumann95) [inline app/models/planning_application.rb:267]: Can this be a private method?

File: app/models/planning_application.rb [hunk 2]
```diff
@@ -261,7 +261,15 @@ def flagged_proposal_details(flag)
...
+    "#{ENV['APPLICANTS_APP_HOST']}.#{local_authority.subdomain}.gov.uk/validation_requests?planning_application_id=#{id}&change_access_id=#{change_access_id}"
+  end
+
+  def internal_url
```
Comment (benbaumann95) [inline app/models/planning_application.rb:271]: private method?

**Review by benbaumann95 — COMMENTED** [review]
If that ENV is set globally then would Lambeth (who don't have the public url set up) get provided with the wrong url?

**Comment by EGiataganas** [comment]
I was thinking that @benbaumann95 but technically we shouldn't have any council that is not ready. So lambeth shouldn't use our service until they are full onboard

**Review by benbaumann95 — COMMENTED** [review]
Is it worth adding a spec for this too ?

**Comment by EGiataganas** [comment]
### BEFORE DEPLOYING
- we need to add the new environment variable on AWS
- we need to change `APPLICANTS_APP_HOST` environment variable on AWS

**Review by benbaumann95 — APPROVED** [review]

**Review by allbecauseyoutoldmeso — APPROVED** [review]

**Pull request merged**